### PR TITLE
UI Customization Options in Startup script

### LINKED ISSUE #151: UI Customization Options in Startup script
To make demos clearer and slightly more custom, lets add some basic customization options to the startup script:

- [x] Setting the name of the application: Default still "Social Web Demo"
- [x] Setting/choosing a color / color pallet?
    - Not sure how this would work, but perhaps this is just changing the header background or something
    - The idea is to make it extra clear in cross provider demos that the application

### PR DESCRIPTION
**Pull request by mattheworris**
# Purpose
To make demos clearer and slightly more custom, lets add some basic customization options to the startup script:

- [x] Setting the name of the application: Default still "Social Web Demo"
- [x] Setting/choosing a color / color pallet?

Closes #151 

## Solution
The `start.sh` script should ask a few questions, save the values in the env file and make sure that those values are passed through to the docker environment and correctly processed by the frontend to change the display.

### Change summary
- Updated `start.sh` to ask for name to appear in header and background color
- Updated `stop.sh` to match `start.sh` updates
- Updated `header.tsx` to get the name and color values from the environment
- Updated `docker-compose.yaml` to pass through the env values
- Add code to redact sensitive values when displaying the env file
- Add code to hide the input of sensitive values

## Steps to Verify
1. Use `start.sh` and start from scratch to create a new env file by answering the questions.
2. Verify the demo reflects the correct changes.

## Additional details / screenshot
This example shows a name change and a light green background color:
Screenshot 2024-08-22 at 10 37 01
Screenshot 2024-08-22 at 12 56 19

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -194,12 +228,21 @@ if [ -f ${ENV_FILE} ]; then
...
+        if [ -f ${ENV_FILE} ]
+        then
+            ${OUTPUT} "Failed to remove previous saved environment. Exiting..."
+        fi
     fi
 fi
 
 if [ ! -f ${ENV_FILE} ]
 then
+    ${OUTPUT} << EOI
```
Comment (mattheworris) [inline start.sh:242]: moved up for clarity

File: start.sh [hunk 2]
```diff
@@ -212,22 +255,21 @@ then
...
-EOI
     echo "COMPOSE_PROJECT_NAME=${COMPOSE_PROJECT_NAME}" >> ${ENV_FILE}
+    echo
     read -p "Enter a tag to use to pull the Gateway Docker images [latest]: " tag
     echo "DOCKER_TAG=${tag:-latest}" >> ${ENV_FILE}
+
     # Ask the user if they want to start on testnet or local
+    echo
     read -p "Do you want to start on Frequency Paseo Testnet [y/N]: "
+    echo
```
Comment (mattheworris) [inline start.sh:266]: adjust some spacing

File: start.sh [hunk 3]
```diff
@@ -123,17 +123,28 @@ ask_and_save() {
...
     local prompt=${2}
     local default_value=${3}
+    local hide_input=${4:-false}
     local value=
+    local input=
+
     if [ -z "${default_value}" ]
     then
-        input=
-        while [ -z "${input}" ]
```
Comment (mattheworris) [inline start.sh:130]: We can allow empty values for ipfs user and secret

File: start.sh [hunk 4]
```diff
@@ -152,6 +163,39 @@ show_help() {
     echo "  -n, --name                 Specify the project name"
 }
 
+
+# Function to get user selection and return the hex value of the selected color
+select_color() {
+    # Define color options
+    declare -a color_names=("White" "Light Gray" "Light Yellow" "Light Blue" "Light Green" )
```
Comment (mattheworris) [inline start.sh:170]: Using indexed arrays to support the older version of Bash that ships with macs. Associative arrays (similar to maps) are not available.

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**